qt/scientific_interfaces/ISISReflectometry/GUI/Save/AsciiSaver.cpp

```diff
@@ -71,28 +59,30 @@ Mantid::API::Workspace_sptr AsciiSaver::workspace(std::string const &workspaceNa
   return ads.retrieveWS<Mantid::API::Workspace>(workspaceName);
 }
 
-Mantid::API::IAlgorithm_sptr AsciiSaver::setUpSaveAlgorithm(std::string const &saveDirectory,
-                                                            const Mantid::API::Workspace_sptr &workspace,
-                                                            std::vector<std::string> const &logParameters,
-                                                            FileFormatOptions const &fileFormat) const {
-  auto const saveAlg = getSaveAlgorithm();
-  auto const extension = extensionForFormat(fileFormat.format());
-  auto const savePath = assembleSavePath(saveDirectory, fileFormat.prefix(), workspace->getName(), extension);
+void AsciiSaver::runSaveAsciiAlgorithm(std::string const &savePath, std::string const &extension,
+                                       const Mantid::API::Workspace_sptr &workspace,
+                                       std::vector<std::string> const &logParameters,
+                                       FileFormatOptions const &fileFormat) const {
+  m_saveAlgRunner->runSaveAsciiAlgorithm(workspace, savePath, extension, logParameters,
+                                         fileFormat.shouldIncludeHeader(), fileFormat.shouldIncludeQResolution(),
+                                         fileFormat.separator());
+}
 
-  saveAlg->setProperty("InputWorkspace", workspace);
-  saveAlg->setProperty("Filename", savePath);
-  saveAlg->setProperty("FileExtension", extension);
-  saveAlg->setProperty("LogList", logParameters);
-  saveAlg->setProperty("WriteHeader", fileFormat.shouldIncludeHeader());
-  saveAlg->setProperty("WriteResolution", fileFormat.shouldIncludeQResolution());
-  saveAlg->setProperty("Separator", fileFormat.separator());
-  return saveAlg;
+void AsciiSaver::runSaveORSOAlgorithm(std::string const &savePath, const Mantid::API::Workspace_sptr &workspace,
+                                      FileFormatOptions const &fileFormat) const {
+  m_saveAlgRunner->runSaveORSOAlgorithm(workspace, savePath, fileFormat.shouldIncludeQResolution());
 }
 
 void AsciiSaver::save(const Mantid::API::Workspace_sptr &workspace, std::string const &saveDirectory,
                       std::vector<std::string> const &logParameters, FileFormatOptions const &fileFormat) const {
-  auto alg = setUpSaveAlgorithm(saveDirectory, workspace, logParameters, fileFormat);
-  alg->execute();
+  auto const extension = extensionForFormat(fileFormat.format());
+  auto const savePath = assembleSavePath(saveDirectory, fileFormat.prefix(), workspace->getName(), extension);
```

Comment:
Could these be made into references (`auto const &&extension`) or if they need to be copies, `std::move`d into the `runSave...()` methods?

Reply:
The double ampersand looks like it's a newer C++ feature and I'm still at an early stage in my understanding of lvalues and rvalues I'm afraid, so would you mind explaining a bit more about why this change would be beneficial? My understanding at the moment is that once we've assigned what I think are rvalue strings returned by `extensionForFormat` and `assembleSavePath` to `extension` and `savePath`, we're then passing them by reference in the rest of the code.